### ISSUE TITLE
NRDS on Windows saving ncpa.cfg

### ISSUE THREAD

**Issue by ejbvanc** [NagiosEnterprises/ncpa#65]
After download but during saving of the NRDS configuration on a Windows client, nrds.py errors out and is unable to save the file.  I believe it is because self.config.file_path is not returning the configuration filename.  This is not an issue on Linux.  Here is the URL for the exact line where the issue is: https://github.com/NagiosEnterprises/ncpa/blob/master/agent/passive/nrds.py#L107

**Comment by scot0357** [comment]
Preliminary testing suggests that this was fixed by fixing some other passive problems (IE the passive agent not able to run plugins). If during testing this doesn't work I'll revisit.

**Comment by lgroschen** [comment]
https://github.com/NagiosEnterprises/ncpa/issues/212

**Comment by jhomann** [comment]
From reading the code, and trying this... I am not even sure how ANY of you got this to even remotely work. It's 100% completely broken. This shouldn't be done the way it is trying to be done.

I may have to push this off until a 2.1.0 release or something because this may take more time than I have at the moment to resolve. However, I think it would be best to make the nrds configs be purely a [passive check] configuration file stored in ncpa.cfg.d instead of overwriting part of ncpa's own config file. We can update other things as we go but this is what I will shoot for in the future. For the time being I think I will remove most documentation and not hide it in the config but warn that it has not yet been implemented.

I'd like to fix the current problems in NCPA before adding on new ones.

**Comment by jomann09** [comment]
A note to myself. When NRDS functionality gets added in we will need to create a way to tell the passive service to refresh the config file. It can probably be done pretty easily while it's running as long. This is due to the change for running plugins where `parse_config` was being ran every time the passive checks were ran. This was probably intended to catch changes that NRDS had made to the config.

**Comment by jm-welch** [comment]
To clarify on your last statement, the passive service (currently - v1.8.1) reads the config each time it triggers, correct? Was a change made that stopped that from occurring?

**Comment by jomann09** [comment]
Yes, the passive agent does not update the config every time it runs. It requires a restart for that to happen. However, if you need it to, there is a way to force the passive service to refresh it's config values. The real annoyance is that it does all the config setup a little different depending on the OS it is running on.
